test/e2e/deployment/configmap.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2022 The SODA Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deployment
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"github.com/google/uuid"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	log "github.com/sirupsen/logrus"
+	"github.com/soda-cdm/kahu/client"
+	"github.com/soda-cdm/kahu/controllers/app/options"
+	k8s "github.com/soda-cdm/kahu/test/e2e/util/k8s"
+	kahu "github.com/soda-cdm/kahu/test/e2e/util/kahu"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+//testcase for E2E deployment backup and restore
+var _ = Describe("ConfigMapBackup", Label("configmap"), func() {
+	Describe("configmapE2ETest", func() {
+		Context("Create backup of configmap and restore", func() {
+			It("configmap", func() {
+
+				optManager, err := options.NewOptionsManager()
+				Expect(err).To(BeNil())
+				if err != nil {
+					log.Fatalf("Failed to initialize controller option manager")
+				}
+				cfg, err := optManager.Config()
+				Expect(err).To(BeNil())
+				if err != nil {
+					log.Errorf("Failed to get configuration %s", err)
+					os.Exit(1)
+				}
+
+				f := client.NewFactory(AgentBaseName, &cfg.KahuClientConfig)
+				kubeClient, err := f.KubeClient()
+				kahuClient, err := f.KahuClient()
+
+				//Create configmap to test
+				ns := "default"
+				labels := make(map[string]string)
+
+				UUIDgen, _ := uuid.NewRandom()
+				name := "configmap" + "-" + UUIDgen.String()
+				configMap, err := k8s.CreateConfigMap(kubeClient, ns, name, labels)
+				log.Infof("configMap:%v\n", configMap)
+				Expect(err).To(BeNil())
+
+				err = k8s.WaitForConfigMapComplete(kubeClient, ns, name)
+				Expect(err).To(BeNil())
+
+				//create backup for the configMap
+				backupName := "backup" + "configmap" + "-" + UUIDgen.String()
+				includeNs := "default"
```

Comment:
To keep testcases independant and without impact to existing system, it is better to create a new namespace for backup and restore and delete them after the execution

Reply:
done